### PR TITLE
[Codegen] NFC: Common the method that is used for adding transform dialect passes.

### PR DESCRIPTION
**Pull request by MaheshRavishankar**

### CODE REVIEW

**Review by antiagainst — APPROVED** [review]

File: compiler/src/iree/compiler/Codegen/Common/CMakeLists.txt
Comment (qedawkins) [inline compiler/src/iree/compiler/Codegen/Common/CMakeLists.txt]: My recent change wound up the build a bit and I think that's the cause for the CI failures you're hitting. The `TransformDialectInterpreter` now relies on `Codegen::Common` (this) because it is pulling in `UserConfig.cpp` to set user configs. That said, `MaterializeUserConfigs.cpp` should be the only file even referencing user configs at this point, so I would just move `UserConfig.cpp` to the cc_library below and reverse the link between `Common` and `TransformDialectInterpreter`.

(Github wouldn't let me comment on the line I wanted to, so I just commented on the closest one I could find)

Comment (qedawkins) [reply compiler/src/iree/compiler/Codegen/Common/CMakeLists.txt]: There is a tangentially related detail here, most likely what we should be doing is requiring a callback function to be provided per-backend that registers all of the required dialects with the TransformInterpreter pass. This way we aren't referencing the NVGPU and AMDGPU dialects in Codegen/Common. I was waiting for some other work to land first though that reworks the interpreter a bit.

Comment (MaheshRavishankar) [reply compiler/src/iree/compiler/Codegen/Common/CMakeLists.txt]: I am confused actually... both these files are in `Codegen/Common`

**Review by hanhanW — APPROVED** [review]

File: compiler/src/iree/compiler/Codegen/Common/MaterializeUserConfigs.cpp
Comment (qedawkins) [inline compiler/src/iree/compiler/Codegen/Common/MaterializeUserConfigs.cpp:43]: nit: make static

**Review by qedawkins — APPROVED** [review]

File: compiler/src/iree/compiler/Codegen/Common/MaterializeUserConfigs.cpp
Comment (MaheshRavishankar) [reply compiler/src/iree/compiler/Codegen/Common/MaterializeUserConfigs.cpp:43]: well, its already in anonymous namespace

Comment (qedawkins) [reply compiler/src/iree/compiler/Codegen/Common/MaterializeUserConfigs.cpp:43]: Oh that's true. If it means rerunning CI don't worry about it, but I find having the static there helpful for reading the code.

**Comment by MaheshRavishankar** [comment]
@qedawkins what we discussed doesnt work either. There is some complex dependency structure here that resulted in circular dependency. I am not using this for now. Ill convert it to draft. When you unwind this a bit, this might be landable.

**Comment by qedawkins** [comment]
> @qedawkins what we discussed doesnt work either. There is some complex dependency structure here that resulted in circular dependency. I am not using this for now. Ill convert it to draft. When you unwind this a bit, this might be landable.

darn :(
I'll fix it after #15340 lands then. The end goal being, everything in common is built at the same time.

**Comment by qedawkins** [comment]
Filed an issue for it here: #15357

**Comment by benvanik** [comment]
Closing stale PR.

**Pull request closed without merging**
